hydromt/models/model.py

```diff
@@ -133,6 +130,38 @@ def __init__(
             f"Initializing {self._NAME} model from {model_metadata['plugin_name']} (v{model_metadata['version']})."
         )
 
+        self._region_component_name = self._determine_region_component(region_component)
+
+    def _determine_region_component(self, region_component: Optional[str]) -> str:
+        if region_component is not None:
+            if region_component not in self._components:
+                self.logger.warning(
+                    f"Component {region_component} not found in components."
+                    "You can add it afterwards with add_component."
+                )
```

Comment:
I wonder if this is behavior we want to encourage. I think after model init all required components should be initialized even if their data may stay empty.
But maybe we should allow for non spatial models (if they exist?) so we can keep this warning but change the message?

Reply:
> I think after model init all required components should be initialized even if their data may stay empty.
Then people will no longer be able to use the Model in ad hoc ways and only plugins can customise the behaviour of the model beyond what we provide at init. Is that what we want?

Reply:
If you are a CLI user, you are unlikely to get this error, as it is common that these users are plugins. Therefore the model should be in place for them.

We might want to make this into an error, but it's quite useful to have this for testing.

Reply:
I will try to make it an error to raise